roadmap/implementers-guide/src/node/disputes/dispute-coordinator.md

```diff
@@ -4,10 +4,324 @@ The coordinator is the central subsystem of the node-side components which parti
+need to preemptively import approval votes, which has shown to be a very
+inefficient process. (Quadratic complexity adds up, with 35 votes in total per candidate)
+
+Approval votes are very relevant non the less as we are going to see in the next
+section.
+
+## Ensuring Malicious Approval Votes Will Be Recorded
+
+While there is no need to record approval votes in the dispute coordinator
+preemptively, we do need to make sure they are recorded when a dispute
+actually happens. This is because only votes recorded by the dispute
+coordinator will be considered for slashing. While the backing group always gets
+slashed, a serious attack attempt will likely also consist of malicious approval
+checkers which will cast approval votes, although the candidate is invalid. If
+we did not import those votes, those nodes would likely cast an `invalid` explicit
+vote as part of the dispute in addition to their approval vote and thus avoid a
+slash. With the 2/3rd honest assumption it seems unrealistic that malicious
+actors will keep sending approval votes once they became aware of a raised
+dispute. Hence the most crucial approval votes to import are the early ones
+(tranche 0), to take into account network latencies and such we still want to
+import approval votes at a later point in time as well (in particular we need to
+make sure the dispute can conclude, but more on that later).
+
+As mentioned already previously, importing votes is most efficient when batched.
+At the same time approval voting and disputes are running concurrently so
+approval votes are expected to trickle in still, when a dispute is already
+ongoing.
+
+Hence, we have the following requirements for importing approval votes:
+
+1. Only import them when there is a dispute, because otherwise we are
+   wasting lots of resources _always_ for the exceptional case of a dispute.
+2. Import votes batched when possible, to avoid quadratic import complexity.
+3. Take into account that approval voting is still ongoing, while a dispute is
+   already running.
+
+With a design where approval voting sends votes to the dispute-coordinator by
+itself, we would need to make approval voting aware of ongoing disputes and
+once it is aware it could start sending all already existing votes batched and
+trickling in votes as they come. The problem with this is, that it adds some
+unnecessary complexity to approval voting and also we might still import most of
+the votes unbatched, but one-by-one, depending on what point in time the dispute
+was raised.
+
+Instead of the dispute coordinator telling approval-voting that a dispute is
+ongoing for approval-voting to start sending votes to the dispute coordinator,
+it would make more sense if the dispute-coordinator would just ask
+approval-voting for votes of candidates that are currently disputed. This way
+the dispute-coordinator can also pick the time when to ask and we can therefore
+maximize the amount of batching.
+
```

Comment:
I had to read this more than once, maybe I'd write something like so: 
Therefore, instead of the dispute coordinator informing approval voting of an ongoing dispute and to begin forwarding *(all?)* votes to the coordinator; it makes more sense for the dispute coordinator to just ask approval voting for the votes of candidates in dispute. This way, the dispute coordinator can also pick the best time for maximizing the number of votes in the batch.

Reply:
Thanks!